satpy/_config.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2016-2019 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""Satpy Configuration directory and file handling."""
+
+import glob
+import logging
+import os
+import sys
+from collections import OrderedDict
+
+import pkg_resources
+from donfig import Config
+import appdirs
+
+LOG = logging.getLogger(__name__)
+
+BASE_PATH = os.path.dirname(os.path.realpath(__file__))
+# FIXME: Use package_resources?
+PACKAGE_CONFIG_PATH = os.path.join(BASE_PATH, 'etc')
+
+_satpy_dirs = appdirs.AppDirs(appname='satpy', appauthor='pytroll')
+_CONFIG_DEFAULTS = {
+    'cache_dir': _satpy_dirs.user_cache_dir,
+    'data_dir': _satpy_dirs.user_data_dir,
+    'config_path': [],
+}
+
+# Satpy main configuration object
+# See https://donfig.readthedocs.io/en/latest/configuration.html
+# for more information.
+#
+# Configuration values will be loaded from files at:
+# 1. The builtin package satpy.yaml (not present currently)
+# 2. $SATPY_ROOT_CONFIG (default: /etc/satpy/satpy.yaml)
+# 3. <python-env-prefix>/etc/satpy/satpy.yaml
+# 4. ~/.config/satpy/satpy.yaml
+# 5. ~/.satpy/satpy.yaml
+# 6. $SATPY_CONFIG_PATH/satpy.yaml if present (colon separated)
+_CONFIG_PATHS = [
+    os.path.join(PACKAGE_CONFIG_PATH, 'satpy.yaml'),
+    os.getenv('SATPY_ROOT_CONFIG', os.path.join('/etc', 'satpy', 'satpy.yaml')),
+    os.path.join(sys.prefix, 'etc', 'satpy', 'satpy.yaml'),
+    os.path.join(os.path.expanduser('~'), '.config', 'satpy', 'satpy.yaml'),
+    os.path.join(os.path.expanduser('~'), '.satpy', 'satpy.yaml'),
+]
```

Comment:
how about appdir's user_config_dir?

Reply:
Turns out on linux `user_config_dir` is equivalent to `~/.config/satpy` so I changed that path to use `user_config_dir` and updated the documentation. So this is done too.